### PR TITLE
fix: Allow to remove the finalizer even if the ScaledObject isn't valid

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/4396

### CODE REVIEW

**Comment by JorTurFer** [comment]
@ppanyukov, 
FYI: I think that this PR solves your issue deleting invalid ScaledObjects approving any update that only removes the finalizers (unlocking the deletion of those resources)

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 ### Fixes
 
+- **General**: Allow to remove the finalizer even if the ScaledObject isn't valid ([#4396](https://github.com/kedacore/keda/issue/4396))
```
Comment (tomkerkhove) [inline CHANGELOG.md:57]: Is this general or related to the validation?

Comment (JorTurFer) [reply CHANGELOG.md:57]: > Is this general or related to the validation?

I don't get your question. Currently, the ScaledObject can't be removed if the validation fails, so I'm not sure. The error is that validation should allow to remove the finalizers, but the ScaledObject is locked at cluster level for everything

Comment (tomkerkhove) [reply CHANGELOG.md:57]: So the change is specific to validation =)

In that case:
```suggestion
- **Admission Webhooks**: Allow to remove the finalizer even if the ScaledObject isn't valid ([#4396](https://github.com/kedacore/keda/issue/4396))
```

**Review by tomkerkhove — APPROVED** [review]
